core/src/main/scala/org/apache/spark/metrics/sink/PrometheusServlet.scala

```diff
@@ -24,15 +24,21 @@ import jakarta.servlet.http.HttpServletRequest
 import org.eclipse.jetty.servlet.ServletContextHandler
 
 import org.apache.spark.SparkConf
+import org.apache.spark.annotation.{DeveloperApi, Since, Unstable}
 import org.apache.spark.ui.JettyUtils._
 
 /**
+ * :: DeveloperApi ::
  * This exposes the metrics of the given registry with Prometheus format.
  *
  * The output is consistent with /metrics/json result in terms of item ordering
  * and with the previous result of Spark JMX Sink + Prometheus JMX Converter combination
  * in terms of key string format.
+ *
+ * This is used by Spark MetricsSystem internally and Spark K8s operator.
  */
+@Unstable
+@DeveloperApi
```

Comment:
Why do we have `PrometheusPullModelSink` in `K8s Operator` repository instead of here, @jiangzho ?

Reply:
To avoid confusion, `PrometheusPullModelSink` in operator has been renamed to `PrometheusPullModelHandler` to indicate that it's implementing `HttpHandler` and serve the metrics over another http port. I was hesitating but it does not provide additional `Sink` functionalities (i.e. not pushing metrics directly like other sinks do). 

When operator is ready to integrate with Prometheus for the push model, we'd definitely start here instead of in operator repo only.